### DISCUSSION TITLE
Melhorias no Cuida para a próxima versão (v3.0.0)

### DISCUSSION THREAD

**Discussion by RafaelGondi** [Sysvale/cuida#268] [Nova versão]
## Discussão sobre as melhorias para a v3.0.0

Galera, vamos listar nessa discussão sugestões de melhoria para serem lançadas na próxima versão do Cuida. Problemas e melhorias cuja correção/implementação envolvam breaking changes e causem incompatibilidades nas APIs dos componentes são importantes de discutirmos aqui.

Também é importante debatermos a migração do DS para as novas tecnologias que a gente vêm discutindo recentemente (Vue3, Vite e TypeScript), visto que elas vão causar breaking changes.

Por último, pontos de melhoria no Design Kit (Cuida no Figma) também são interessantes de serem trazidos aqui.

Essa discussão ficará aberta por 1 mês (até 01/09), momento a partir do qual passaremos a debater questões relativas à implementação.
As sugestões acatadas serão posteriormente convertidas em issues.

Caso vocês tenham novas sugestões, objeções, observações, etc. é só colocar abaixo. o/

**Comment by RafaelGondi** [comment]
Remover o bootsrap-vue como dependência de projeto

**Reply by RafaelGondi** [reply]
- Adiciona no readme aviso para que novos componentes não utilizem bootstrap-vue
- Adiciona nos templates de PR, e issue aviso para que novos componentes não utilizem bootstrap-vue

**Reply by RafaelGondi** [reply]
- Substituir o b-table pela tabela do Cuida na documentação
- Remover b-buttons e b-nav

**Comment by RafaelGondi** [comment]
Remover Feather Icons e material-icons como dependências de projeto

**Comment by RafaelGondi** [comment]
Remover o componente [Dropdown](https://sysvale.github.io/cuida/?path=/docs/componentes-forms-dropdown--drop-down), visto que já temos o [ClusteredMultiselect ](https://sysvale.github.io/cuida/?path=/docs/componentes-forms-clusteredmultiselect--clustered-multiselect)como componente com o mesmo propósito e ele está atualizado com os tokens do Cuida.

**Reply by tassiocaique** [reply]
Sou à favor, mas acredito que esse `ClusteredMultiselect` deveria ser renomeado.

**Comment by RafaelGondi** [comment]
Migrar os testes de Jest para [Vitest](https://vitest.dev/) A documentação do Vue está recomendando o uso do Vitest em projetos Vite:
![image](https://user-images.githubusercontent.com/20057968/184129500-c29b3f18-9961-430d-8cb2-f72757de783d.png)

E como a nova versão usa Vite e o Vitest é compatível com o Jest, acho que valeria o teste

**Reply by RafaelGondi** [reply]
-V3

**Reply by RafaelGondi** [reply]
Implementado em: https://github.com/Sysvale/cuida/pull/286

**Comment by johnn1sbo3s** [comment]
Padronizar os textos da documentação (novamente)

**Comment by johnn1sbo3s** [comment]
Ajustar variantes de componentes que não estão funcionando corretamente no Figma. _Exemplo: botões_

**Reply by RafaelGondi** [reply]
- Botões: resolvido
- Mapear componentes com conflito de props / variantes

**Comment by johnn1sbo3s** [comment]
Discutir sobre a tipografia utilizada no Figma.

**Reply by RafaelGondi** [reply]
- Analisar novas possibilidades de fonte
- Substituir a fonte no CS
- Refatorar tokens de tipografia

**Comment by tassiocaique** [comment]
Discutir a padronização das apis dos componentes. Se vai usar `is-[NOME]` para props booleanas, etc.

Obs.: Não sou a favor do "is-", foi só um exemplo.

**Reply by RafaelGondi** [reply]
Massa! Acho que a gente pode adicionar essas guidelines no CONTRIBUTING.MD e no Desing Kit.

**Reply by RafaelGondi** [reply]
- V3

**Comment by johnn1sbo3s** [comment]
Discutir sobre a mudança de DOs DON'Ts na documentação.

**Reply by RafaelGondi** [reply]
### Exemplos com DOs e DON'Ts: 
- [Alma](https://zeroheight.com/7a44c04f2/v/0/p/482c4c-input/b/09d7b1) **Design System BR**

![image](https://user-images.githubusercontent.com/20057968/184392198-b2872110-cb64-46fe-af1c-9365e9f1e40d.png)

- [Ocean](https://zeroheight.com/9c9b2b3aa/p/089a1f-button/b/19fdad) **Design System BR**

![image](https://user-images.githubusercontent.com/20057968/184416035-af948ff0-3342-4a08-9e11-845a198a37d6.png)

- [Vtex Styleguide](vtex.styleguide.com)

![image](https://user-images.githubusercontent.com/20057968/184388476-cd47b303-00e0-476a-bef4-f188077e1c0c.png)

**Reply by RafaelGondi** [reply]
### Outras maneiras de abordar:
- [Bold](https://bold.bridge.ufsc.br/pt/components/button) **Design System BR**

![image](https://user-images.githubusercontent.com/20057968/184391438-4d8ac28b-2197-4c0e-b2d5-14ee7c034cf8.png)

- [Garden](https://garden.zendesk.com/components/anchor)

![image](https://user-images.githubusercontent.com/20057968/184388324-48781db2-b1e2-4a1b-a5c4-8ef34312537d.png)

- [Sirius](https://sirius.zenvia.com/docs/components/checkbox/)

![image](https://user-images.githubusercontent.com/20057968/184415837-1080ed21-0974-4fe8-8dfe-7ba9d52479c3.png)

- [Spectrum](https://spectrum.adobe.com/page/badge/)

![image](https://user-images.githubusercontent.com/20057968/184388941-edd9baef-1804-491c-b590-fa0a950e6f65.png)

**Reply by RafaelGondi** [reply]
### Híbrido:
- [Jota DS](https://zeroheight.com/561dd07ed/v/latest/p/948862-button/b/950c00) **Design System BR**

![image](https://user-images.githubusercontent.com/20057968/184390062-e11e3023-e209-4ad5-a37d-b59fe2e6fb7e.png)

- [Gestalt](https://gestalt.netlify.app/web/button)

![image](https://user-images.githubusercontent.com/20057968/184389479-84a27479-1d0a-4319-9808-afe684ce4f18.png)

**Reply by RafaelGondi** [reply]
Eu curto a ideia dos DOs e DON'Ts. É um padrão de documentação já bem difundido e que deixa menos rígido o formato de escrita

**Reply by johnn1sbo3s** [reply]
Perfeito. Eu também gosto da ideia de DOs e DONT's. Acho que ajudaria a gente a ter mais liberdade na forma como indica as recomendações pra cada componente. O que pode ser negativo aí, além de utilizar termos em inglês, é essa liberdade de escrever de qualquer forma acabar discrepante demais de um componente pra outro.

**Reply by GeissyMay** [reply]
Eu estava pensando na estrutura do design system do Nimbus, mas pensei em adicionarmos o tópico de boas práticas que eu vi no design system do Bold.
Ficaria: 
- Quando usar?
- Quando não usar?
- Boas práticas
![image](https://user-images.githubusercontent.com/110190001/185191563-79be4991-05f2-473e-8201-3ec3ac943358.png)
![image](https://user-images.githubusercontent.com/110190001/185191603-a3c6ee2f-0cf1-47a3-9ef8-1187c045c2c2.png)

**Reply by RafaelGondi** [reply]
Esse modelo é legal, mas o "Quando usar / não usar" tem o mesmo problema do modelo que usamos hoje. Ele limita a escrita do que ver a seguir, como descrito no exemplo acima, a resposta tem que vir com "Quando precisamos....", "Quando o feedback...", "Quando necessitamos...", "Quando queremos...", etc

**Reply by GeissyMay** [reply]
Não acho que limite no mesmo nível. Pensei nessa forma pq deixa claro tanto quando usar quanto algumas informações a mais de boas práticas. Eu acho mais importante não mudarmos tanto a estrutura hoje e pensarmos mais em reformular o conteúdo/escrita. E não gosto muito do termo em inglês em uma documentação em português, pode ter em alguns design systems e ser reconhecido esse método mas não concordo.

**Reply by RafaelGondi** [reply]
Tu pode dar alguns exemplos de como a limitação ocorreria em nível diferente? 

Com relação a mudar a estrutura, acredito que, caso achemos necessário, o momento de mudança é agora que estamos indo pra uma breaking change, indo pra versão 3.0 do projeto. 

Sobre termos em inglês na documentação em português, além desse caso ser comum na indústria, também é bem comum utilização de termos e jargões em inglês por causa da nossa área (computação e design). Além disso, todos os nossos componentes, convenções de código e tokens, são em inglês. A tendência é que o projeto como um todo passe a ser em inglês no futuro.

**Reply by GeissyMay** [reply]
A estrutura do tópico/frase anterior "Recomendamos o uso {...} **quando**" pede, necessariamente, no início de cada tópico subsequente, a presença de algumas palavras específicas como "For", "Se", "Houver", "Estiver". Essa "obrigatoriedade" por sua vez condiciona o restante da frase a um formato um pouco mais engessado.

A utilização do "Quando usar" não coloca a necessidade de se iniciar o tópico subsequente dentro dos exemplos anteriores, o que, de certa forma, amplia o leque de possibilidades para frases mais simples, curtas e/ou claras.

Exemplo:

Quando usar

- Visualizar status associados com a lógica de negócios ou ações do usuário
- Precisar categorizar algo
- Mostrar conteúdos de Readonly

Quando não usar

- O badge puder ser removido pelo usuário
- Precisar clicar no componente para executar uma ação ou funcionalidade

**Reply by RafaelGondi** [reply]
Não entendi bem. Tomando o exemplo acima:
"Quando usar

Visualizar status associados com a lógica de negócios ou ações do usuário"

A gente tem dois verbos juntos (usar e visualizar) e a frase fica "Quando usar visualizar status...", de modo que me parece que esse formato exige a mesma estrutura, ou estrutura semelhante ao atualmente utilizado

**Comment by johnn1sbo3s** [comment]
Atualizar estilo do componente Badge no Figma.

**Reply by RafaelGondi** [reply]
PR aberto: https://github.com/Sysvale/cuida/pull/288

**Comment by johnn1sbo3s** [comment]
Criar tokens de container

**Reply by johnn1sbo3s** [reply]
Tokens de container para centralizar o conteúdo das páginas. 
Hoje, o conteúdo é alinhado às margens, o que quebra a responsividade em telas maiores.

Exemplo:
![image](https://user-images.githubusercontent.com/49643446/187539440-5f15ada2-1c72-4f7a-99f6-b7286b29f752.png)

A ideia é que tenhamos o conteúdo centralizado em telas com maior resolução, como faz, por exemplo,a página do Bootstrap:

Zoom em 100%:
![image](https://user-images.githubusercontent.com/49643446/187540174-f870327e-7022-4f70-b748-ad33242456b0.png)

Zoom em 67%:
![image](https://user-images.githubusercontent.com/49643446/187540273-e8b34a7b-29f5-4477-8967-f7b9a8e12c20.png)

**Reply by johnn1sbo3s** [reply]
Exemplo de tokens do Bootstrap:

![image](https://user-images.githubusercontent.com/49643446/187540470-05d851ff-6e40-44ca-8921-8f31c62d8297.png)

**Comment by johnn1sbo3s** [comment]
Atualizar o componente Table no Figma para o padrão.

**Comment by johnn1sbo3s** [comment]
Criar componente de abas no Figma.

**Reply by RafaelGondi** [reply]
- Feito
